docs/cookbook.md

```diff
@@ -0,0 +1,412 @@
+```bash
+bin/console doctrine:schema:update --force
+```
+
+You can check that the table is correctly created in your database.
+ 
+## Your reference data as an attribute
+Now declare the new reference data in the application : 
+```yaml
+#app/config/config.yml
+pim_reference_data:
+    supplier:
+        class: Acme\Bundle\SupplierBundle\Entity\Supplier
+        type: simple
+```
+
+To check every thing is OK so far, execute the following commands :
+```bash
+#check if the entity doctrine configuration is OK
+bin/console doctrine:mapping:info
+```
+```bash
+#check if the reference data is correct
+bin/console pim:reference-data:check
+```
+
+At this point, the reference data can already be used as an attribute in your families. Try to create a new "reference data simple select" attribute.
+
+## Reference data CRUD
+
+### Menu
+A new menu item can be added to access the reference data list. To achieve that, declare the menu item by creating a new configuration file :
+```yaml
+#Acme\Bundle/SupplierBundle/Resources/config/form_extensions/menu.yml
+extensions:
+    acme-menu-supplier:
+        module: pim/menu/item
+        parent: pim-menu-reference_data-navigation-block
+        position: 10
+        config:
+            title: acme.menu.item.reference_data.supplier
+            to: pim_customentity_index
+            routeParams:
+                customEntityName: supplier
+```
+At this point the item menu should be visible at the end of the menu. You may have to clear your cache (including the front end cache) to see it.
+
+### Datagrid configuration
+
+Create a new configuration file to declare the datagrid configuration for supplier. First, declare the source and columns configuration just to have a basic list page : 
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/datagrid/supplier.yml
+datagrid:
+    supplier:
+        options:
+            entityHint: supplier
+            manageFilters: false
+        source:
+            type: pim_datasource_default
+            entity: Acme\Bundle\SupplierBundle\Entity\Supplier
+            repository_method: createDatagridQueryBuilder
+        columns:
+            code:
+                label: acme.supplier.field.code
+            name:
+                label: acme.supplier.field.name
+```
+
+#### Form extension
+
+Declare the reference data index page form extension. Exemple of working configuration : https://github.com/akeneo-labs/CustomEntityBundle/blob/master/docs/examples/CustomBundle/Resources/config/form_extensions/brand/index.yml
+
+Declare your new custom entity :
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/custom_entities.yml
+custom_entities:
+    supplier:
+        entity_class: Acme\Bundle\SupplierBundle\Entity\Supplier
+```
+
+At this point, you should have a basic empty list of reference data, with no sort ou filter available. A create button may be visible, but not configured yet.
```

Comment:
In fact, I would add this part before the form extension.
Then, add the form extension with create ref data part.
It allows you to create easily data you will filter and sorter on.